test/e2e/serial/tests/configuration.go

```diff
@@ -343,5 +345,137 @@ var _ = Describe("[serial][disruptive][slow] numaresources configuration managem
+			wg.Wait()
+
+			By("checking that NodeResourceTopology objects available resources has changed")
+			for _, initialNrt := range initialNrtList.Items {
+				wg.Add(1)
+				go func(nrt *nrtv1alpha1.NodeResourceTopology) {
+					defer GinkgoRecover()
+					defer wg.Done()
+					Eventually(func() (bool, error) {
+						updatedNrt := &nrtv1alpha1.NodeResourceTopology{}
+						err = fxt.Client.Get(context.TODO(), client.ObjectKeyFromObject(nrt), updatedNrt)
+						Expect(err).ToNot(HaveOccurred())
+						// since the tests are not running in parallel, we know that nothing can pollute
+						// the current NRT object, and therefore it is guaranteed that the difference between
+						// the initial and updated NRT object derived from the change in the reserved CPUs under the
+						// kubeletconfig. we don't care from the exact accounting, we just want to make sure that there is
+						// a difference.
+						equal, err := e2enrt.CheckEqualAvailableResources(*nrt, *updatedNrt)
+						if err != nil {
+							klog.Warningf("failed to check equal available resources of NodeResourceTopology %q", updatedNrt.Name)
+							return false, err
+						}
+						if equal == true {
+							klog.Warningf("NodeResourceTopology %q has not been updated after kubelet modification", updatedNrt.Name)
+							return false, nil
+						}
+						return true, nil
+					}, timeout, time.Second*30).Should(BeTrue())
+				}(&initialNrt)
+			}
+			wg.Wait()
+
+			defer func() {
+				By("reverting kubeletconfig changes")
+				for i := 0; i < len(targetedKCs); i++ {
+					kcAsMc := targetedKCs[i]
+					err = fxt.Client.Get(context.TODO(), client.ObjectKeyFromObject(kcAsMc), kcAsMc)
+					Expect(err).ToNot(HaveOccurred())
+
+					kcObj, err := kubeletconfig.MCOKubeletConfToKubeletConf(kcAsMc)
+					Expect(err).ToNot(HaveOccurred())
+
+					kcObj.ReservedSystemCPUs = initialRsvCPUs[kcAsMc.Name]
+					err = kubeletconfig.KubeletConfToMCKubeletConf(kcObj, kcAsMc)
+					Expect(err).ToNot(HaveOccurred())
+
+					err = fxt.Client.Update(context.TODO(), kcAsMc)
+					Expect(err).ToNot(HaveOccurred())
+				}
+
+				By("waiting for MachineConfigPools to get updated")
+				for _, mcp := range mcps {
+					wg.Add(1)
+					go func(mcpool *machineconfigv1.MachineConfigPool) {
+						defer GinkgoRecover()
+						defer wg.Done()
+						err = e2ewait.ForMachineConfigPoolCondition(fxt.Client, mcpool, machineconfigv1.MachineConfigPoolUpdated, configuration.MachineConfigPoolUpdateInterval, configuration.MachineConfigPoolUpdateTimeout)
+						Expect(err).ToNot(HaveOccurred())
+					}(mcp)
+				}
+				wg.Wait()
+			}()
+		})
 	})
 })
+
+// we want to make sure that we're changing the value to be something different from the existing one.
+// since we don't know the current ReservedSystemCPUs values, let's pick something random, make sure it's different and apply it.
+func applyNewReservedSystemCPUsValue(oldRsvCPUs *string) {
+	newRsvCPUs := "3-4"
+	// if it happens to be the same, pick something else
```

Comment:
correct, but we should iterate a bit over it and no just once, should'nt we?
also, this deserves a log

Reply:
> but we should iterate a bit over it and no just once

Not sure I understand. why?

Reply:
in the code above we try a couple CPUs to reserve. If this fails, we try another set (determined how?) and then we give up.
Why only two attempts? We may want to either iterate more over CPU pairs to use as reserved, or to document why only two attempts is enough.

Reply:
Each KubeletConfig has a (single) field if the ```ReservedSystemCPUs```
In runtime, we don't know the current value of this field, but actually, we don't care what the actual value is. All we want it to assign a value that is different from the existing one, in order to trigger a machine-config change.

Since we don't care about the value, we just pick something totally random (```newRsvCPUs := "3-4"```) now since we don't know the current value, there are two options here:
1. the current value is equal to the random value we choose
2. the current value is **not** equal to the random value we choose.

In option number 2 we are good to go, but if happened and we land on option number one, it won't trigger a machine-config change (because the value has left the same) so we just pick another random value, which now we are certain that it is different from the existing one.

To conclusion, the maximum attempts is 2.

Reply:
ok, makes sense even if it still feels clunky somehow. But I don't mind. Please distill this comment into a source code comment, and we can move forward

Reply:
Is the comment above the function signature isn't enough?

Reply:
you can just extend that comment explaining why trying twice is sufficient